src/Reporter.ts

```diff
@@ -242,6 +252,11 @@ class Reporter {
                 ? request = https.request(this.requestOptions, responseHandler)
                 : request = http.request(this.requestOptions, responseHandler);
 
+            request.setTimeout(REPORTER_HTTP_TIMEOUT, () => {
+                ThundraLogger.debug('<Reporter> Reporter request timeout');
```

Comment:
I think, log level should be at error level and log message should mention about possible public internet connectivity issue

```js
ThundraLogger.error('<Reporter> Reporter socket timeout. Please be sure that your application has access to public internet.');
```